### PR TITLE
Add views to jenkins dashboards

### LINKED ISSUE #168: Discard old builds and sort the jobs per view.
### Is your feature request related to a problem? Please describe

Currently the anonymous role in public jenkins shows all workflows. Create a new view for external community to show only relevant workflows

### Describe the solution you'd like
Can be categorized into 4 sections to start with:

- All : (present by default) exactly what we have today https://build.ci.opensearch.org/
- Build : Will contain all build pipelines which are distribution builds, docker builds as well as gradle checks, etc
- Test: Will contain all test pipelines example integ-test, perf-test, bwc-tests, etc
- Release: Will contain all release pipelines including promotion jobs, client releases, etc.

### Describe alternatives you've considered

Keep it as it is. 

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by gaiksaya**
### Description
This change adds views feature to Jenkins.
Views on Jenkins dashboard allows us to classify jobs into different sections. By enabling views, all the jobs will be classified into 4 categories namely Build, Test, Release and Misc. 
* Build - Any job containing word `build` in it.
* Test - Any job containing word `test` in it.
* Release - Any job containing word `release` in it.
* Misc - All the jobs that belong to neither of the above categories.

### Issues Resolved
resolves #168 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: lib/compute/views.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      list: {
+        columns: [
+          'status',
+          'weather',
+          'jobName',
+          'lastSuccess',
+          'lastFailure',
+          'lastDuration',
+          'buildButton',
+          'favoriteColumn',
```
Comment (peterzhuamazon) [inline lib/compute/views.ts:21]: What is this for in the view?

File: test/data/test_env.yaml [hunk 2]
```diff
@@ -200,6 +248,12 @@ tool:
               installers:
                 - adoptOpenJdkInstaller:
                     id: jdk-19.0.1+10
+      - name: openjdk-20
+        properties:
+          - installSource:
+              installers:
+                - adoptOpenJdkInstaller:
+                    id: jdk-20.0.1+9
```
Comment (peterzhuamazon) [inline test/data/test_env.yaml:256]: You should rebase as this is very old commit.

File: lib/compute/views.ts (hunk 1 above)
Comment (gaiksaya) [reply lib/compute/views.ts:21]: Columns displayed. I have kept what was default.

File: test/data/test_env.yaml (hunk 2 above)
Comment (gaiksaya) [reply test/data/test_env.yaml:256]: I did still showed up. Looks like tests weren't updated for a while.

File: lib/compute/views.ts (hunk 1 above)
Comment (peterzhuamazon) [reply lib/compute/views.ts:21]: Got it.

File: test/data/test_env.yaml (hunk 2 above)
Comment (peterzhuamazon) [reply test/data/test_env.yaml:256]: Oh I see, wonder how it is not being detected prior.

**Comment by peterzhuamazon** [comment]
Hi @gaiksaya should we upgrade cdk as mend is reporting?

**Review by peterzhuamazon — APPROVED** [review]

**Comment by gaiksaya** [comment]
> Hi @gaiksaya should we upgrade cdk as mend is reporting?

That might be a breaking change to upgrade the cdk version. Will take it up in later PR.
Thanks!

File: test/data/test_env.yaml (hunk 2 above)
Comment (gaiksaya) [reply test/data/test_env.yaml:256]: I believe because that part of the yaml is not being checked / verified by any test

**Pull request merged**